app/lib/features/spaces/widgets/space_list_widget.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:acter/common/toolkit/errors/error_page.dart';
+import 'package:acter/common/widgets/room/room_card.dart';
+import 'package:acter/features/space/widgets/space_sections/section_header.dart';
+import 'package:acter/features/spaces/providers/space_list_provider.dart';
+import 'package:acter/features/spaces/widgets/space_list_skeleton.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::space-list-widget');
+
+class SpaceListWidget extends ConsumerWidget {
+  final String? searchValue;
+  final int? limit;
+  final bool showSectionHeader;
+  final VoidCallback? onClickSectionHeader;
+  final bool shrinkWrap;
+  final Widget emptyState;
+
+  const SpaceListWidget({
+    super.key,
+    this.limit,
+    this.searchValue,
```

Comment:
I wanted to make `SpaceListWidget` independent of any params. So `searchValue` shouldn't be mandatory in that way. Yes, in current scenario `searchValue` is given to this widget but this not how generic widget I am thinking off. It can adopt to any use-case.